File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.java
class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        int n = words.length;
        int minDistance = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            if (words[i].equals(target)) {
                int linearDist = Math.abs(i - startIndex);
                int circularDist = Math.min(linearDist, n - linearDist);
                minDistance = Math.min(minDistance, circularDist);

                if (minDistance == 0) {
                    return 0;
                }
            }
        }

        return minDistance == Integer.MAX_VALUE ? -1 : minDistance;
    }
}

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: list[str], target: str, startIndex: int) -> int:
        n = len(words)
        min_dist = float('inf')

        for i, word in enumerate(words):
            if word == target:
                direct_dist = abs(i - startIndex)
                circular_dist = min(direct_dist, n - direct_dist)
                min_dist = min(min_dist, circular_dist)

        return min_dist if min_dist != float('inf') else -1